extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/dataconnection/MongoDataConnection.java

```diff
@@ -186,7 +187,10 @@ private static void addResources(List<DataConnectionResource> resources, MongoDa
     @Override
     public void destroy() {
         if (mongoClient != null) {
```

Comment:
It's already covered by https://github.com/hazelcast/hazelcast/pull/24355

Reply:
Reverted https://github.com/hazelcast/hazelcast/pull/24358/commits/4bdfdc4e44a45c024607b3b70d882e93f5e98aa7